File: oauth2-boot2-legacy/src/main/java/samples/oauth2/service/TokenBeautifier.java
package samples.oauth2.service;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.apache.commons.codec.binary.Base64;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.io.IOException;
import java.util.Map;

@Service
public class TokenBeautifier {

    @Autowired
    private ObjectMapper objectMapper;

    public String formatJwtToken(String token) {
        try {
            return toPrettyJsonString(parseToken(token));
        } catch (Exception e) {
            return "";
        }
    }

    private Map<String, ?> parseToken(String base64Token) throws IOException {
        String token = base64Token.split("\\.")[1];
        return this.objectMapper.readValue(Base64.decodeBase64(token),
                new TypeReference<Map<String, ?>>() {});
    }

    private String toPrettyJsonString(Object object) throws Exception {
        return this.objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(object);
    }
}

File: applications/oauth2-boot2-native/src/main/kotlin/sample/oauth2/service/TokenBeautifier.kt
package sample.oauth2.service

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import com.nimbusds.jose.util.Base64
import org.springframework.stereotype.Service

@Service
class TokenBeautifier(val objectMapper: ObjectMapper) {

    fun formatJwtToken(token: String): String {
        return toPrettyJsonString(parseToken(token))
    }

    private fun parseToken(base64Token: String): Map<String, *> {
        val tokens = base64Token.split(".")
        val token = tokens[1]
        return this.objectMapper.readValue(Base64(token).decodeToString())
    }

    private fun toPrettyJsonString(obj: Any): String {
        return this.objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(obj)
    }
}
